### PR TITLE
Fix SSL tests

### PR DESCRIPTION
**Pull request by michaelgoin**
## Proposed Release Notes

* Modified self-signed SSL cert to use 'localhost' instead of 'ssl.lvh.me' for SSL testing.

* Removed unnecessary trace observer configuration validation for host and port.

## Details

The 'ssl.lvh.me' domain expired. Given our style of testing, using 'localhost' appears to work fine. As such, updated the self-signed cert generation to use 'localhost' and updated tests accordingly.

The host validation for the trace observer was preventing usage of 'localhost'. Early in infinite tracing implementation, there used to be a single field that would get parsed out and I even believe there might have been a path included at one point. grpc-js (at the time) would crash when things were invalid so we had extra protections.

These aren't needed now as the fields are split, the port defaults and the host is given as copy-paste text to customers. Anything else invalid will fail to connect.

We could have some safeguards but most will likely be overkill and not something we typically do (i.e. port).

### CODE REVIEW

File: test/integration/collector-remote-method.tap.js
Comment (bizob2828) [inline test/integration/collector-remote-method.tap.js:16]: thoughts on moving this `SSL_HOST` to a test constant that can be referenced in unit, integration, versioned tests?

File: test/lib/fake-collector.js
Comment (bizob2828) [inline test/lib/fake-collector.js:18]: does this need to get updated?

File: bin/ssl.sh
Comment (bizob2828) [inline bin/ssl.sh:70]: learned the hard way, you have to delete self signed cert and ca-index if they already exist, which they prob do

```
rm test/lib/self-signed-test-certificate.crt test/lib/ca-index
```

**Review by bizob2828 — APPROVED** [review]
overall looks good, glad it was as easy as changing host of self signed cert.  I did have a food for thought about centralizing the host to a test constant in case in the future we don't have to go through this effort and just flip one var across the repo.  I will approve but wait for your thoughts in lieu of a merge

File: test/integration/collector-remote-method.tap.js
Comment (michaelgoin) [reply test/integration/collector-remote-method.tap.js:16]: Hopefully would never be used by the unit tests haha (yes they already do bad thing slike call example.com).

I considered it but I decided the tradeoff wasn't currently worth it. But I'm not strong against.

Just seemed like plumbing the constant into a central location that potentially adds another file dependency wasn't really worth it for how few places were using. I guess that SSL helper could return the constant at some point and that wouldn't be too bad. I just like to be wary of adding new problems to avoid duplication (but yes don't want to update 1000 files when you have to make a change either).

Now that I'm done thinking out loud, I'm fine either way.

File: test/lib/fake-collector.js
Comment (michaelgoin) [reply test/lib/fake-collector.js:18]: I had that thought but seems like it is just returned from the preconnect and not really used? Certainly nothing broke and collector-1.lvh.me didn't have a cert or anything. That being said, could be updated to get rid of the .lvh to reduce confusion.

File: bin/ssl.sh
Comment (michaelgoin) [reply bin/ssl.sh:70]: `./bin/ssl.sh clear` also does the trick.

File: test/lib/fake-collector.js
Comment (bizob2828) [reply test/lib/fake-collector.js:18]: looking at this var and the code around checking host and `ACTUAL_HOST`, this should no longer get hit and is dead code

Comment (bizob2828) [reply test/lib/fake-collector.js:18]: i'll take care of this since you had to leave unexpectedly

File: test/integration/collector-remote-method.tap.js
Comment (bizob2828) [reply test/integration/collector-remote-method.tap.js:16]: yea my thought here was this PR was quite involved to change the host in a cert and its corresponding tests that use the cert

Comment (bizob2828) [reply test/integration/collector-remote-method.tap.js]: i made these changes but don't have perms to push to your fork, so ill just open a separate PR after I merge this

**Review by bizob2828 — APPROVED** [review]

**Review by carlo-808 — APPROVED** [review]
👍

**Pull request merged**
